src/main/java/uk/gov/hmcts/divorce/caseworker/service/ReIssueApplicationService.java

```diff
@@ -0,0 +1,126 @@
+package uk.gov.hmcts.divorce.caseworker.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.divorce.caseworker.service.task.GenerateCitizenRespondentAosInvitation;
+import uk.gov.hmcts.divorce.caseworker.service.task.GenerateMiniApplication;
+import uk.gov.hmcts.divorce.caseworker.service.task.GenerateRespondentSolicitorAosInvitation;
+import uk.gov.hmcts.divorce.caseworker.service.task.SendAosNotifications;
+import uk.gov.hmcts.divorce.caseworker.service.task.SendAosPack;
+import uk.gov.hmcts.divorce.caseworker.service.task.SendApplicationIssueNotifications;
+import uk.gov.hmcts.divorce.caseworker.service.task.SetPostIssueState;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.ReissueOption;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+
+import java.time.Clock;
+
+import static java.time.LocalDate.now;
+import static uk.gov.hmcts.divorce.divorcecase.task.CaseTaskRunner.caseTasks;
+
+@Service
+@Slf4j
+public class ReIssueApplicationService {
+
+    @Autowired
+    private SetPostIssueState setPostIssueState;
+
+    @Autowired
+    private GenerateMiniApplication generateMiniApplication;
+
+    @Autowired
+    private GenerateRespondentSolicitorAosInvitation generateRespondentSolicitorAosInvitation;
+
+    @Autowired
+    private GenerateCitizenRespondentAosInvitation generateCitizenRespondentAosInvitation;
+
+    @Autowired
+    private SendAosPack sendAosPack;
+
+    @Autowired
+    private SendApplicationIssueNotifications sendApplicationIssueNotifications;
+
+    @Autowired
+    private SendAosNotifications sendAosNotifications;
+
+    @Autowired
+    private Clock clock;
+
+    @Value("${resissue.days-until-overdue}")
+    private Integer daysUntilOverdue;
+
+    public CaseDetails<CaseData, State> process(final CaseDetails<CaseData, State> caseDetails) {
+        ReissueOption reissueOption = caseDetails.getData().getApplication().getReissueOption();
+
+        log.info("For case id {} reissue option selected is {} ", caseDetails.getId(), reissueOption.getLabel());
+
+        caseDetails.getData().getApplication().setReissueDate(now(clock));
+
+        CaseDetails<CaseData, State> updatedCaseDetails = null;
```

Comment:
it's a minor point, but when I see code like this (a switch statement that sets a variable) I think it's slightly cleaner to move the switch statement into a method and then use return values, so this would be:

```
var updatedCaseDetails = updateCase(reissueOption, details);
```